File: examples/helloworld-java/src/main/java/org/yanex/telegram/example/TelegramProperties.java
package org.yanex.telegram.example;

import org.jetbrains.annotations.NotNull;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Properties;

public class TelegramProperties {
    private static final String FILE_NAME = "telegram.properties";
    
    private static final String PROP_TOKEN = "token";
    private static final String PROP_LAST_ID = "lastId";
    
    @NotNull
    public String getToken() {
        String token = readProperties().getProperty(PROP_TOKEN);
        if (token == null) {
            throw new RuntimeException("Token is not set in " + FILE_NAME);
        }
        return token;
    }
    
    public void setLastId(long lastId) {
        Properties props = readProperties();
        props.setProperty(PROP_LAST_ID, Long.toString(lastId));
        writeProperties(props);
    }
    
    public long getLastId() {
        String lastId = readProperties().getProperty(PROP_LAST_ID);
        return lastId == null ? 0 : Long.parseLong(lastId);
    }
    
    private static void writeProperties(Properties props) {
        try (FileOutputStream output = new FileOutputStream(FILE_NAME)) {
            props.store(output, null);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
    
    private static Properties readProperties() {
        Properties props = new Properties();
        try (FileInputStream input = new FileInputStream(FILE_NAME)) {
            props.load(input);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        return props;
    }
}

File: examples/helloworld-kotlin/src/main/java/org/yanex/telegram/example/TelegramProperties.kt
package org.yanex.telegram.example

import java.io.File
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.FileOutputStream
import java.util.Properties

class TelegramProperties(val fileName: String = "telegram.properties") {
    val token: String
        get() = readProperties().getProperty(PROP_TOKEN) ?: throw RuntimeException("Token is not set in " + fileName)

    var lastId: Long
        get() = readProperties().getProperty(PROP_LAST_ID)?.let { it.toLong() } ?: 0
        set(lastId) = writeProperties(readProperties().apply {
            setProperty(PROP_LAST_ID, lastId.toString())
        })

    private fun writeProperties(props: Properties) {
        FileOutputStream(fileName).use { output -> props.store(output, null) }
    }

    private fun readProperties() = Properties().apply {
        try {
            FileInputStream(fileName).use { input -> load(input) }
        } catch (e: FileNotFoundException) {
            throw FileNotFoundException("File ${System.getProperty("user.dir")}${File.separator}$fileName not found.")
        }
    }

    companion object {
        val PROP_TOKEN = "token"
        val PROP_LAST_ID = "lastId"
    }
}
